### PR TITLE
Define a new outputErrorPage() function in HTTPErrorExceptions

### LINKED ISSUE #254: HTTP exception must not echo HTML contents directly
Currently, HTTP exception classes output HTML content directly on construction. This does not separate concerns.

Suggest a method `$e->outputErrorPage()` on the exception class to output the error page, which will be called by the dispatcher.

### PR DESCRIPTION
**Pull request by bugos**
Fixes #254

### CODE REVIEW

File: run [hunk 1]
```diff
@@ -20,6 +20,9 @@
     try {
         exit( $controller->dispatch( [ 'method' => $method ], $vars, [], 'POST' ) );
     }
+    catch ( HTTPErrorException $e ) {
+        die('An HTTP error occured: '. $e->header);
```
Comment (dionyziz) [inline run]: nit: spaces inside `( )`; you need a single space after `(` and a single space before `)`

Comment (dionyziz) [inline run]: :s/occured/occurred

**Comment by vitsalis** [comment]
LGTM

File: run [hunk 2]
```diff
@@ -20,6 +20,9 @@
     try {
         exit( $controller->dispatch( [ 'method' => $method ], $vars, [], 'POST' ) );
     }
+    catch ( HTTPErrorException $e ) {
+        die( 'An HTTP error occuρred: '. $e->header );
```
Comment (dionyziz) [inline run]: You seem to have an extraneous UTF-8 character `ρ` here.

File: run [hunk 3]
```diff
@@ -20,6 +20,9 @@
     try {
         exit( $controller->dispatch( [ 'method' => $method ], $vars, [], 'POST' ) );
     }
+    catch ( HTTPErrorException $e ) {
+        die( 'An HTTP error occurred: '. $e->header );
```
Comment (dionyziz) [inline run]: nit: Space before `.`

**Pull request merged**